### PR TITLE
Avoid using fx.Interpreter in nvfuser executor function

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Using fx.Interpreter is a nice way of modifying the calls inside of FX graphs, but it introduces unnecessary overhead in this case.

Example:
```py
import torch
from torch.fx.experimental.proxy_tensor import make_fx
from torch._prims.context import TorchRefsNvfuserCapabilityMode
from torch._prims.executor import execute
a = torch.randn(3, 2, dtype=torch.float16, device="cuda")
s = torch.sigmoid
d = torch.digamma  # digamma is not supported in nvfuser and aten eager execution is used
def func(a):
    return s(d(s(d(s(d(s(a)))))))
with TorchRefsNvfuserCapabilityMode():
    gm = make_fx(func)(a)

%%timeit
execute(gm, a, executor="nvfuser"); torch.cuda.synchronize();
# On master: 350 µs
# With this PR: 130 µs
```

### CODE REVIEW

File: torch/_prims/nvfuser_executor.py [hunk 1]
```diff
@@ -171,6 +171,15 @@ def maybe_partition_graph(gm: GraphModule):
...
             )
         partitioned_graph = partitioner.fuse_partitions(partitions)
+
+        # Overwriting graph's __call__() function with nvfuser_execute
+        # This avoids the need to call the interpreter on the graph
+        for node in partitioned_graph.graph.nodes:
+            # TODO: use a better way to identify fused submodule
+            if node.op == "call_module" and "fused_" in node.name:
+                fused_module = getattr(partitioned_graph, node.name)
+                fused_module._wrapped_call = nvfuser_execute
```
Comment (ezyang) [inline torch/_prims/nvfuser_executor.py:181]: Don't do this. https://github.com/pytorch/torchdynamo/blob/042c574479c58cf68a5c100f0e08e7a86eb87fb3/torchdynamo/optimizations/training.py#L268-L321 shows the correct way to put in an opaque module that calls into the executor, and https://github.com/pytorch/torchdynamo/blob/042c574479c58cf68a5c100f0e08e7a86eb87fb3/torchdynamo/optimizations/training.py#L362-L371 shows how to do the graph replacement.

Comment (IvanYashchuk) [reply torch/_prims/nvfuser_executor.py]: Thanks! That's cleaner.

File: torch/_prims/nvfuser_executor.py [hunk 2]
```diff
@@ -171,6 +180,17 @@ def maybe_partition_graph(gm: GraphModule):
                 category=RuntimeWarning,
             )
         partitioned_graph = partitioner.fuse_partitions(partitions)
+
+        # Overwriting graph's fused submodules with a wrapper module with
+        # __call__() method that calls nvfuser_execute.
+        # This avoids the need to call the interpreter on the graph
```
Comment (ezyang) [inline torch/_prims/nvfuser_executor.py:186]: out of date now

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**